### PR TITLE
Introduce workdays calculation support

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Introduce workdays related calculation support in Python Holidays. Add following public methods to `HolidayBase`:
- `is_workday(dt)` - check if date is a working day
- `get_nth_workday(dt, n)` - return n-th working day from date
- `get_workdays_number(dt1, dt2)` - return the number of working days between two dates

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [x] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: holidays/holiday_base.py [hunk 1]
```diff
@@ -746,14 +749,14 @@ def _add_special_holidays(self, mapping_names, observed=False):
...
                     self._add_holiday(
                         self.tr(self.substituted_label)
-                        % date(
-                            optional[0] if optional else self._year, from_month, from_day
-                        ).strftime(self.tr(self.substituted_date_format)),
+                        % from_date.strftime(self.tr(self.substituted_date_format)),
                         to_month,
                         to_day,
                     )
+                    self.workdays.add(from_date)
```
Comment (arkid15r) [inline holidays/holiday_base.py:759]: I'm not sure if it's worth it to put an explicit _is_weekend check before adding date into weekend_workdays.
```suggestion
                    self.weekend_workdays.add(from_date)
```

**Review by arkid15r — COMMENTED** [review]
Great PR, this will make UX more consistent with users' expectations.

Thank you!

File: holidays/holiday_base.py [hunk 2]
```diff
@@ -944,6 +947,36 @@ def get_named(
...
+            dt1, dt2 = dt2, dt1
+
+        return sum(
+            self.is_workday(dt1 + timedelta(days=n)) for n in range(1, (dt2 - dt1).days + 1)
+        )
+
+    def is_workday(self, key: DateLike) -> bool:
+        """Return True if date is a working day (not a holiday or a weekend)."""
+        dt = self.__keytransform__(key)
+        return not (dt in self or (self._is_weekend(dt) and (dt not in self.workdays)))
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/holiday_base.py]: Done.

File: holidays/holiday_base.py (hunk 1 above)
Comment (KJhellico) [reply holidays/holiday_base.py]: I think such a check would be a bit redundant. 👀

**Review by arkid15r — DISMISSED** [review]
LGTM 👍

**Review by arkid15r — APPROVED** [review]

**Pull request merged**